vnext/Microsoft.ReactNative.Cxx/JSValue.cpp

```diff
@@ -349,6 +349,11 @@ JSValueObject JSValueObject::Copy() const noexcept {
   return object;
 }
 
+JSValue &JSValueObject::operator[](std::string_view propertyName) noexcept {
+  auto result = try_emplace(propertyName.data(), nullptr);
```

Comment:
I don't see the definition for try_emplace, but is this treating the `string_view` like a null-terminated C string? The end of a string_view isn't always null-terminated (e.g. you can have a view over just the beginning of a string), so that could potentially do the wrong thing.

Reply:
Oh, I see now that we're privately inheriting from std::map. This would mean this line can read things outside of the string_view.

Reply:
Good point! Let me see how to make it better. I would like to avoid creating a string copy when we lookup data.

Reply:
Are we not doing that right now? I might be missing something but I was assuming the above would implicitly construct a `std::string`.

Reply:
I just made a commit with a fix. The std::less<> comparison algorithm, which we define in the base class for JSValueObject, works in a way that when it compares stored std::string key with a std::string_view, it converts std::string to std::string_view first, and thus no extra string allocation. We only need to convert std::string_view to std::string when create an entry.